pkg/build/api/types.go

```diff
@@ -476,3 +479,12 @@ type BuildLogOptions struct {
 	// is not available yet. Otherwise the server will wait until the build has started.
 	NoWait bool
 }
+
+// SecretSpec specifies a secret to be included in a build pod and its corresponding mount point
+type SecretSpec struct {
```

Comment:
Sorry Clayton, I was thinking more of custom builds than anything else. The others needs more thought :)

It's clearly useful to have access to secrets during Source and Docker builds, but the actual mechanics of it are ugly. For Source builds I guess I could put them in a volume that I mount into the container so the assemble script can read them but not include them in the committed image. For Docker builds, I would have to copy them into the source and then make the build responsible for deleting them.

For custom builds, are you saying that we could just let you specify the list of VolumeSources and VolumeMounts you want in your pod and be done with it?